test/schemes/fluid/surface_normal_sph.jl

```diff
@@ -0,0 +1,269 @@
+    # Test case 1: Simple linear arrangement of particles
+    nparticles = 5
+    particle_spacing = 1.0
+    NDIMS = 2
+
+    coordinates = zeros(NDIMS, nparticles)
+    for i in 1:nparticles
+        coordinates[:, i] = [i * particle_spacing, 0.0]
+    end
+    velocity = zeros(NDIMS, nparticles)
+    mass = fill(1.0, nparticles)
+    fluid_density = 1000.0
+    density = fill(fluid_density, nparticles)
+
+    system, bnd_system, semi, ode = create_fluid_system(coordinates, velocity, mass,
+                                                        density,
+                                                        particle_spacing;
+                                                        NDIMS=NDIMS)
+
+    compute_and_test_surface_normals(system, semi, ode; NDIMS=NDIMS)
+end
+
+@testset "Sphere Surface Normals" begin
+    # Test case 2: Particles arranged in a disk
+    particle_spacing = 0.25
+    radius = 1.0
+    center = (0.0, 0.0)
+    NDIMS = 2
+
+    # Create a `SphereShape`, which is a disk in 2D
+    sphere_ic = SphereShape(particle_spacing, radius, center, 1000.0)
+
+    coordinates = sphere_ic.coordinates
+    velocity = zeros(NDIMS, size(coordinates, 2))
+    mass = sphere_ic.mass
+    density = sphere_ic.density
+
+    # To get somewhat accurate normals we increase the smoothing length unrealistically
+    system, bnd_system, semi, ode = create_fluid_system(coordinates, velocity, mass,
+                                                        density,
+                                                        particle_spacing;
+                                                        NDIMS=NDIMS,
+                                                        smoothing_length=3.0 *
+                                                                         particle_spacing)
+
+    compute_and_test_surface_normals(system, semi, ode; NDIMS=NDIMS)
+
+    nparticles = size(coordinates, 2)
+    expected_normals = zeros(NDIMS, nparticles)
+    surface_particles = Int[]
+
+    # Compute expected normals and identify surface particles
+    for i in 1:nparticles
+        pos = coordinates[:, i]
+        r = pos .- center
+        norm_r = norm(r)
+
+        # If particle is on the circumference of the circle
+        if abs(norm_r - radius) < particle_spacing
+            expected_normals[:, i] = -r / norm_r
+
+            push!(surface_particles, i)
+        else
+            expected_normals[:, i] .= 0.0
+        end
+    end
+
+    # Normalize computed normals
+    computed_normals = copy(system.cache.surface_normal)
+    for i in surface_particles
+        norm_computed = norm(computed_normals[:, i])
+        if norm_computed > 0
+            computed_normals[:, i] /= norm_computed
+        end
+    end
+
+    # Compare computed normals to expected normals for surface particles
+    @test isapprox(computed_normals[:, surface_particles],
+                   expected_normals[:, surface_particles], norm=x -> norm(x, Inf),
+                   atol=0.05)
```

Comment:
```suggestion
    @test isapprox.(computed_normals[:, surface_particles],
                   expected_normals[:, surface_particles], atol=0.05)
```
My earlier suggestions were stupid.

Reply:
Not sure what you try here? But this is wrong.

Reply:
Why is it wrong? Instead of comparing two vectors with a specific norm, we can just pointwise apply `isapprox` to check every element of the vector.

Reply:
I meant
```suggestion
    @test all(isapprox.(computed_normals[:, surface_particles],
                   expected_normals[:, surface_particles], atol=0.05))
```
```
julia> isapprox(x, y, atol=10eps(), norm=x -> norm(x, Inf))
true

julia> isapprox(x, y, atol=10eps())
false

julia> all(isapprox.(x, y, atol=10eps()))
true
```

Reply:
Sphere Surface Normals: Error During Test at /media/svenb/Extreme SSD/GIT/TrixiParticles3/test/schemes/fluid/surface_normal_sph.jl:169
  Expression evaluated to non-Boolean